qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/project/QuteProjectRegistry.java

```diff
@@ -81,6 +84,32 @@ private static void registerPrimitiveType(String type) {
 		primitiveTypeArray.setIterableOf(type);
 		javaPrimitiveTypes.put(primitiveTypeArray.getSignature(),
 				CompletableFuture.completedFuture(primitiveTypeArray));
+
+		String objectType = getPrmitiveObjectType(type);
+		if (objectType != null) {
+			autoboxing.put(type, type);
+			autoboxing.put(objectType, type);
+		}
+	}
+
+	private static String getPrmitiveObjectType(String type) {
+		switch (type) {
+		case "boolean":
+			return "java.lang.Boolean";
+		case "byte":
+			return "java.lang.Byte";
+		case "char":
+			return "java.lang.Character";
+		case "double":
+			return "java.lang.Double";
+		case "float":
+			return "java.lang.Float";
+		case "int":
+			return "java.lang.Integer";
+		case "long":
+			return "java.lang.Long";

```

Comment:
maybe reference constants from LiteralSupport.java

Reply:
Very good suggestion, I have moved the function in LiteralSupport class. fixed.